contracts/adapters/AaveMigrationExtension.sol

```diff
@@ -0,0 +1,615 @@
+        }
+
+        // Encode migration parameters for flash loan callback
+        bytes memory params = abi.encode(_decodedParams);
+
+        // Request flash loan for the underlying token
+        POOL.flashLoanSimple(
+            address(this),
+            address(underlyingToken),
+            _underlyingLoanAmount,
+            params,
+            0
+        );
+
+        // Return remaining underlying token to the operator
+        underlyingOutputAmount = _returnExcessUnderlying();
+    }
+
+    /**
+     * @dev Callback function for Aave V3 flash loan, executed post-loan. It decodes the provided parameters, conducts the migration, and repays the flash loan.
+     * @param asset The asset borrowed in the flash loan.
+     * @param amount The amount borrowed.
+     * @param premium The additional fee charged for the flash loan.
+     * @param initiator The initiator of the flash loan.
+     * @param params Encoded migration parameters.
+     * @return True if the operation is successful.
+     */
+    function executeOperation(
+        address asset,
+        uint256 amount,
+        uint256 premium,
+        address initiator,
+        bytes calldata params
+    )
+        external
+        override
+        returns (bool) 
+    {
+        require(msg.sender == address(POOL), "MigrationExtension: invalid flashloan sender");
+        require(asset == address(underlyingToken), "MigrationExtension: invalid flashloan asset");
+        require(initiator == address(this), "MigrationExtension: invalid flashloan initiator");
+
+        // Decode parameters and migrate
+        DecodedParams memory decodedParams = abi.decode(params, (DecodedParams));
+        _migrate(decodedParams);
+
+        underlyingToken.approve(address(POOL), amount + premium);
+        return true;
+    }
+
+    /**
+     * @notice Receives ERC721 tokens, required for Uniswap V3 LP NFT handling.
+     * @dev Callback function for ERC721 token transfers, enabling the contract to receive Uniswap V3 LP NFTs. Always returns the selector to indicate successful receipt.
+     * @return The selector of the `onERC721Received` function.
+     */
+    function onERC721Received(
+        address, // operator
+        address, // from
+        uint256, // tokenId
+        bytes calldata // data
+    ) 
+        external
+        override
+        returns (bytes4)
+    {
+        return this.onERC721Received.selector;
+    }
+
+    /**
+     * @notice OPERATOR ONLY: Transfers any residual balances of the underlying token, Aave's 
+     * wrapped token, and the wrapped SetToken to the operator's address.
+     * @dev This function is intended to recover tokens that might have been left behind
+     * due to the migration process or any other operation. It ensures that the contract
+     * does not retain any assets inadvertently. Only callable by the operator.
+     */
+    function sweepTokens() external onlyOperator {
```

Comment:
Maybe add a method to sweep an arbitrary token from the contract (including eth) ? (mistaken transfers, unexpected dust, future airdrop eligiblity)
Optional though.

Reply:
refactored here [6cc5172](https://github.com/IndexCoop/index-coop-smart-contracts/pull/162/commits/6cc5172a12142b144d74b32f549d4f9b0079e437)